effekt/shared/src/main/scala/effekt/Typer.scala

```diff
@@ -520,13 +528,13 @@ object Typer extends Phase[NameResolved, Typechecked] {
       }
 
       bindings
-  } match { case res => Context.annotateInferredType(pattern, sc); res }
+  } match { case res => Context.annotateInferredType(pattern, List(sc)); res } // TODO MRV: List(sc)
 
   //</editor-fold>
 
   //<editor-fold desc="statements and definitions">
 
-  def checkStmt(stmt: Stmt, expected: Option[ValueType])(using Context, Captures): Result[ValueType] =
+  def checkStmt(stmt: Stmt, expected: Option[List[ValueType]])(using Context, Captures): Result[List[ValueType]] =
```

Comment:
I think this should be a list of option since each individual component might be known or not.